File: python/scrabble-score/scrabble.py
score_mpa = {
    'aeioulnrst': 1,
    'dg': 2,
    'bcmp': 3,
    'fhvwy': 4,
    'k': 5,
    'jx': 8,
    'qz': 10,
}

SCORE_MAP = {c: v
             for k, v in score_mpa.items()
             for c in k}


def score(value):
    if not value.isalpha():
        value = ''
    return sum([SCORE_MAP[c] for c in value.lower()])

File: scala/scrabble-score/src/main/scala/Scrabble.scala
/**
  *
  * @author Cedric Zhuang
  *
  */
class Scrabble {
  private val rawScoreMap: Map[String, Int] = Map(
    "aeioulnrst" -> 1,
    "dg" -> 2,
    "bcmp" -> 3,
    "fhvwy" -> 4,
    "k" -> 5,
    "jx" -> 8,
    "qz" -> 10)

  private lazy val scoreMap: Map[Char, Int] =
    rawScoreMap.flatMap { case (s, v) => s.map(_ -> v) }

  def scoreLetter(c: Char): Int = scoreMap(c.toLower)

  def scoreWord(word: String): Int = word.map(scoreLetter).sum
}
